pkg/apis/postgres-operator.crunchydata.com/v1beta1/shared_types.go

```diff
@@ -149,6 +152,66 @@ func (spec *VolumeClaimSpec) AsPersistentVolumeClaimSpec() corev1.PersistentVolu
 	return out
 }
 
+// VolumeClaimSpecWithAutoGrow extends VolumeClaimSpec with options for
+// automatic volume growth.
+// +structType=atomic
+type VolumeClaimSpecWithAutoGrow struct {
+	VolumeClaimSpec `json:",inline"`
+
+	// +optional
+	AutoGrow *AutoGrowSpec `json:"autoGrow,omitempty"`
+}
+
+// AutoGrowSpec provides options to tune volume auto-growing behavior.
+// Auto grow requires that a limit be set on the PVC.
+type AutoGrowSpec struct {
+	// Trigger is the percentage of used space at which to trigger a volume
+	// expansion.
+	// +optional
+	// +kubebuilder:default=75
+	// +kubebuilder:validation:Minimum=50
+	// +kubebuilder:validation:Maximum=90
+	Trigger *int32 `json:"trigger,omitempty"`
+
+	// MaxGrow is the maximum size to which the volume can be automatically
+	// expanded. If not set, the volume will grow by 50% of the original size each
+	// time the Trigger threshold is exceeded.
+	// +optional
+	MaxGrow *resource.Quantity `json:"maxGrow,omitempty"`
+}
+
+func (spec *AutoGrowSpec) Default() {
```

Comment:
🤔 Do we need this?

Reply:
I'll make a note to check on this later.